addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -0,0 +1,825 @@
+		if (!isConnected()) {
+			logger.debug("No connection to SqueezeServer, will attempt to reconnect now...");
+			connect();
+			if (!isConnected()) {
+				logger.error(
+						"Failed to reconnect to SqueezeServer, unable to send command {}",
+						command);
+				return;
+			}
+		}
+		logger.debug("Sending command: {}", command);
+		try {
+			BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(
+					clientSocket.getOutputStream()));
+			writer.write(command + NEW_LINE);
+			writer.flush();
+		} catch (IOException e) {
+			logger.error("Error while sending command to Squeeze Server ("
+					+ command + ")", e);
+		}
+	}
+
+	/**
+	 * Connects to a SqueezeBox Server
+	 */
+	private void connect() {
+		SqueezeBoxServerConfig config = getConfigAs(SqueezeBoxServerConfig.class);
+		this.host = config.ipAddress;
+		this.cliport = config.cliport;
+		this.webport = config.webport;
+
+		try {
+			clientSocket = new Socket(host, cliport);
+		} catch (IOException e) {
+			logger.error("Failed to connect to the Squeeze Server at "
+					+ config.ipAddress + ":" + config.cliport, e);
+			return;
+		}
+
+		try {
+			listener = new SqueezeServerListener();
+			listener.start();
+		} catch (IllegalThreadStateException e) {
+			logger.error("Failed to start the Squeeze Server listener thread",
+					e);
+			return;
+		}
+
+		updateStatus(ThingStatus.ONLINE);
+		logger.info("Squeeze Server connection started.");
+	}
+
+	/**
+	 * Disconnects from a SqueezeBox Server
+	 */
+	private void disconnect() {
+		// if (!isConnected())
+		// return;
+
+		try {
+			listener.terminate();
+			clientSocket.close();
+		} catch (IOException e) {
+			logger.error("Error attempting to disconnect from Squeeze Server",
+					e);
+			return;
+		} finally {
+			clientSocket = null;
+			listener = null;
+		}
+		players.clear();
+		updateStatus(ThingStatus.OFFLINE);
+		logger.info("Squeeze Server connection stopped.");
+	}
+
+	private class SqueezeServerListener extends Thread {
+		private boolean terminate = false;
+
+		public SqueezeServerListener() {
+			super("Squeeze Server Listener");
+		}
+
+		public void terminate() {
+			logger.warn("Squeeze Server listener being terminated");
+			this.terminate = true;
+		}
+
+		@Override
+		public void run() {
+			logger.debug("Squeeze Server listener started.");
+
+			BufferedReader reader = null;
+			try {
+				reader = new BufferedReader(new InputStreamReader(
+						clientSocket.getInputStream()));
+
+				requestPlayers();
+				sendCommand("listen 1");
+
+				String message;
+				while (!terminate && (message = reader.readLine()) != null) {
+					logger.debug("Message received: {}", message);
+
+					if (message.startsWith("listen 1"))
+						continue;
+
+					if (message.startsWith("players 0")) {
+						handlePlayersList(message);
+					} else {
+						handlePlayerUpdate(message);
+					}
+				}
+			} catch (IOException e) {
+				logger.error("Error in Squeeze Server listener", e);
```

Comment:
when the listener is dying here, you might need some better error handling - the Thing is still online and appears to be connected. Should you set it to OFFLINE and disconnect? Should you retry the connection?